src/librustdoc/doctest.rs

```diff
@@ -321,7 +339,7 @@ fn run_test(
         (true, false) => {}
         (false, true) => {
             if !error_codes.is_empty() {
-                error_codes.retain(|err| !out.contains(&format!("error[{}]: ", err)));
+                error_codes.retain(|err| !out.contains(&format!("error[{}]", err)));
```

Comment:
I think it's because of the added terminal color codes. But it's been so long since I wrote this code that I can't remember for sure...

Reply:
Ok, if it looks fine with the `:` can you change it back?

Reply:
Can you double check why this breaks with `:` added, then add that as a comment?

Reply:
Sure!

Reply:
So with the following file:

```rust
/// You are supposed to call it like this
///
/// # Example
///
/// ```compile_fail,E0599
/// let x = 12i32;
/// x.divo();
/// ```
pub struct Example {}
```

When I run it:

```
$ ./build/x86_64-unknown-linux-gnu/stage1/bin/rustdoc --test foo.rs | cat -e
$
running 1 test$
test foo.rs - Example (line 5) ... FAILED$
$
failures:$
$
---- foo.rs - Example (line 5) stdout ----$
^[[0m^[[1m^[[38;5;9merror[E0599]^[[0m^[[0m^[[1m: no method named `divo` found for type `i32` in the current scope^[[0m$
^[[0m ^[[0m^[[0m^[[1m^[[38;5;12m--> ^[[0m^[[0mfoo.rs:7:3^[[0m$
^[[0m  ^[[0m^[[0m^[[1m^[[38;5;12m|^[[0m$
^[[0m^[[1m^[[38;5;12m4^[[0m^[[0m ^[[0m^[[0m^[[1m^[[38;5;12m| ^[[0m^[[0mx.divo();^[[0m$
^[[0m  ^[[0m^[[0m^[[1m^[[38;5;12m| ^[[0m^[[0m  ^[[0m^[[0m^[[1m^[[38;5;9m^^^^^[[0m^[[0m ^[[0m^[[0m^[[1m^[[38;5;9mmethod not found in `i32`^[[0m$
$
^[[0m^[[1m^[[38;5;9merror^[[0m^[[0m^[[1m: aborting due to previous error^[[0m$
$
^[[0m^[[1mFor more information about this error, try `rustc --explain E0599`.^[[0m$
Some expected error codes were not found: ["E0599"]$
$
failures:$
    foo.rs - Example (line 5)$
$
test result: FAILED. 0 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out$
$
```

It doesn't find the error code because of the terminal color code characters. (termcap, my old nemesis)

So no, we need to remove the ": " part. :)

Reply:
Ok great, please say that as a comment.
```suggestion
                // only look for `error[...]` since there may be color escape characters before and after
                error_codes.retain(|err| !out.contains(&format!("error[{}]", err)));
```